csg.ts

```diff
@@ -185,8 +186,11 @@ CSG.cube = function(options) {
   ].map(function(info) {
     return new CSG.Polygon(info[0].map(function(i) {
       var pos = new CSG.Vector(
+        //@ts-ignore
         c.x + r[0] * (2 * !!(i & 1) - 1),
+        //@ts-ignore
         c.y + r[1] * (2 * !!(i & 2) - 1),
+        //@ts-ignore
         c.z + r[2] * (2 * !!(i & 4) - 1)
```

Comment:
that part `!!(i & 1)` from the initial code seems fishy and typescript complains
add `//@ts-ignore` to make it transpile
to be investigated later

Reply:
How about this?

```
let dX = i & 1 ? r[0] : -r[0]
let dY = i & 2 ? r[1] : -r[1]
let dZ = i & 4 ? r[2] : -r[2]
let pos = new CSG.Vector(c.x + dX, c.y + dY, c.z + dZ)
```

This is why I dislike 'clever' code by the way :). Sooner or later, something like TypeScript will complain.

Reply:
great! fc10b2302107d5039e91ddf6837094957ac4de87